src/ragger/backend/speculos.py

```diff
@@ -187,26 +191,36 @@ def compare_screen_with_snapshot(self,
             return screenshot_equal(f"{golden_snap_path}", snap)
 
     def get_current_screen_content(self) -> dict:
-        self._screen_content = self._retrieve_client_screen_content()
-        return self._screen_content
+        return self._retrieve_client_screen_content()
 
     def compare_screen_with_text(self, text: str) -> bool:
         return text in dumps(self._retrieve_client_screen_content())
 
     def wait_for_screen_change(self, timeout: float = 10.0) -> None:
         start = time()
-        content = self._retrieve_client_screen_content()
-        while content == self._screen_content:
+        screenshot = BytesIO(self._client.get_screenshot())
+        while screenshot_equal(screenshot, self._last_screenshot):
             # Give some time to other threads, and mostly Speculos one
             sleep(0.2)
             if (time() - start > timeout):
                 raise TimeoutError("Timeout waiting for screen change")
-            content = self._retrieve_client_screen_content()
+            screenshot = BytesIO(self._client.get_screenshot())
 
         # Speculos has received at least one new event to redisplay the screen
         # Wait a bit to ensure the event batch is received and processed by Speculos before returning
         sleep(0.2)
 
-        # Update self._screen_content to use it as reference for next calls
-        self._screen_content = self._retrieve_client_screen_content()
-        self.logger.info(f"Received new screen event '{self._screen_content}'")
+        # Update self._last_screenshot to use it as reference for next calls
+        self._last_screenshot = BytesIO(self._client.get_screenshot())
+
+    def wait_for_home_screen(self, timeout: float = 10.0) -> None:
+        screenshot = BytesIO(self._client.get_screenshot())
+        if screenshot_equal(screenshot, self._home_screenshot):
+            return
+
+        endtime = time() + timeout
+        while 1:
+            self.wait_for_screen_change(endtime - time())
+            screenshot = BytesIO(self._client.get_screenshot())
+            if screenshot_equal(screenshot, self._home_screenshot):
+                return
```

Comment:
Potential race if the screen is changed between `!screenshot_equal()` and `self.wait_for_screen_change(endtime - time())`

To solve it make the first `screenshot_equal` comparison with `self._last_screenshot` or write `self._last_screenshot`